paddle/pten/kernels/hybird/cuda/elementwise/elementwise_broadcast.cu.h

```diff
@@ -131,7 +131,7 @@ struct DimensionsTransform {
                                     int i,
                                     int num) {
       for (int j = 1; j < num; ++j) {
-        equal = (in_dims[0][i] == in_dims[j][i]) ? true : false;
+        equal &= (in_dims[0][i] == in_dims[j][i]) ? true : false;
```

Comment:
Before bug fix, Cannot correctly works  in ternary conditions :
```
input1.dims = [2, 5, 2] 
input2.dims = [2, 1, 1] 
input3.dims = [2, 5, 2] 
```
The `equal` param is set without any effect of  input2.dims